lib/cocoapods-core/specification/set.rb

```diff
@@ -68,7 +68,7 @@ def specification_paths_for_version(version)
       #         from highest to lowest.
       #
       def versions
-        versions_by_source.values.flatten.uniq.sort.reverse
+        @versions ||= versions_by_source.values.flatten.uniq.sort.reverse
```

Comment:
This could be invalidated by a core user who got versions, caused a spec update, and gets versions again

Reply:
actually nvm we already cache versions_by_source, this should be fine.

Reply:
Dang. Can you think of any way around that?